### ISSUE TITLE
[ETH-flow] Disable a possibility to place a Buy order

### ISSUE THREAD

**Issue by elena-zh** [cowprotocol/cowswap#1455]
ETH-flow contract does not support Buy orders, so each Buy order transaction fails. 
It would be great to disable a possibility to create a Buy order on the FE side. 

Options: 
1. like we do for SC wallets when we try to buy ETH: 
![image.png](https://images.zenhubusercontent.com/5f55ec970a2e890ec10f7778/20459ca5-c10e-4215-b939-7706640d2046)

2. Disable 'New ETH flow' flag when the buy order with ETH is selected
![image.png](https://images.zenhubusercontent.com/5f55ec970a2e890ec10f7778/480358dc-589d-4748-9235-a9946a072fdd)

**Comment by anxolin** [comment]
Yes! I would do the 2 options (message, and button to change to WETH)
Although is possible we don't need this, because the backend might give support in the future. 

Technically the protocol accepts it, but there are some problems with the access lists. @nlordell knows more about it. 

If this doesn't make it to the next roadmap planning, I would say we should fix it asap in the frontend

**Comment by anxolin** [comment]
Good point! we didn't account for this!

I think we should just change the behaviour of editing the BUY AMOUNT
- So it doesn't change the order into a buy order
- We still estimate how much we need to sell in order to get at least that
- We update the sell amount

The UX is not great because in this flow, the surplus will be in buy token, but I think is the best we can do for the launch. 

WDYT?

**Comment by alfetopito** [comment]
> Good point! we didn't account for this in the planning! (adding it now)
> 
> I think we should just change the behaviour of editing the BUY AMOUNT
> 
>     * So it doesn't change the order into a buy order
> 
>     * We still estimate how much we need to sell in order to get at least that
> 
>     * We update the sell amount
> 
> 
> The UX is not great because in this flow, the surplus will be in buy token, but I think is the best we can do for the launch.
> 
> WDYT?

Hmmm not sure about that.
I think the most simple and explicit way is to disable the swap button if the user changes the receive amount.
Otherwise it's changing the implicit behaviour which is not already that well explained in the first place.

**Comment by anxolin** [comment]
> I think the most simple and explicit way is to disable the swap button if the user changes the receive amount.

Fine, if that's easier we can just disable

> Otherwise it's changing the implicit behaviour which is not already that well explained in the first place.

True, although not sure how important is this, other UIs already work like that. However I agree is not nice to behave so different to normal orders with not explicit change. We could do my approach together with a companion banner, saying ETH Flow orders don't have buy orders, and this is a sell order which mean that any extra token received by price improvements will be received in buy tokens.

Anyways, your approach is kind of nice, but if we add a banner explaining they need to edit the sell amount to get out of this situation
